tarsier/tag_utils.ts

```diff
@@ -397,25 +392,37 @@ function removeNestedTags(elementsToTag: HTMLElement[]): HTMLElement[] {
   return res;
 }
 
+function getTagSymbol(el: HTMLElement): "#" | "$" | "@" | "%" | "" {
+  if (isInteractable(el)) {
+    if (isTextInsertable(el)) {
+      return "#";
+    } else if (el.tagName.toLowerCase() === "a") {
+      return "@";
+    } else {
+      return "$";
+    }
+  } else if (isImageElement(el)) {
+    return "%";
+  } else {
+    return "";
+  }
+}
+
 function insertTags(
   elementsToTag: HTMLElement[],
   tagLeafTexts: boolean,
-): { [key: number]: TagMetadata } {
+): TagMetadata[] {
   function trimTextNodeStart(element: HTMLElement) {
     // Trim leading whitespace from the element's text content
     // This way, the tag will be inline with the word and not textwrap
     // Element text
-    if (!element.firstChild || element.firstChild.nodeType !== Node.TEXT_NODE) {
-      return;
+    if (element.firstChild && element.firstChild.nodeType === Node.TEXT_NODE) {
+      const textNode = element.firstChild as Text;
+      textNode.textContent = textNode.textContent!.trimStart();
```

Comment:
Should keep the guard clause: https://www.codementor.io/@clintwinter/use-guard-clauses-for-cleaner-code-1rrsczgwxp

Also would try to avoid these minor refactors in large PRs since they get in the way